### PR TITLE
Fixed IDENTITY-5183, IDENTITY-5185

### PR DESCRIPTION
**Pull request by isharak**

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java [hunk 1]
```diff
@@ -185,9 +196,16 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
 
             //TODO:This parameter is a custom one and we have to remove if we can collect the user name by having
             // some authentication mechanism.
-            registrationRequestProfile
-                    .setOwner((String) jsonData.get(RegistrationRequest.RegisterRequestConstant.EXT_PARAM_OWNER));
-
+            String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            if(StringUtils.isBlank(username)){
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java]: Check formatting.

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java [hunk 2]
```diff
@@ -185,9 +196,16 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
...
+                Object objOwner = jsonData.get(RegistrationRequest.RegisterRequestConstant.EXT_PARAM_OWNER);
+                if(objOwner != null){
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java]: Check formatting.

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java (hunk 1 above)
Comment (isharak) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java]: Done

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java (hunk 2 above)
Comment (isharak) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java]: done

**Pull request merged**